### ISSUE TITLE
[master] Overhaul of assembly methods

### ISSUE THREAD

**Issue by sebglane** [j507/RotatingMHD#139]
- Fix treatment of source term
- Fix treatment of boundary integrals
- Differentiate assembly in case of coupled or uncoupled problems

**Comment by j507** [comment]
What do you mean by fix in the first two points? Is there an formulation error?

**Comment by sebglane** [comment]
In my opinion, using the variable `gamma` and previous values of the source term and the boundary values is wrong. The source term and the right-hand side should simply be evaluated at the sought-for time level (`time_stepping.get_next_time()`).

**Comment by j507** [comment]
By deviating from the scheme VSIMEX I fear that the temporal stability can't no longer be assured.  Nonetheless, we have already done this with the pressure-correction scheme and the advection term. As long as the tests/Benchmarks can be reproduced with the change, it's okay with me.

**Comment by sebglane** [comment]
Yes, I can understand your point. The thing with the evaluation of the right-hand side (source term, boundary term) is that from the perspective of a fractional step scheme like CNAB it should be evaluated at both time levels, whereas for a BDF scheme it should be evaluated at the desired time.
